sympy/parsing/latex/lark/latex.lark

```diff
@@ -0,0 +1,270 @@
+SYMBOL.1: /[a-zA-Z]/
+SUBSCRIPTED_SYMBOL: /([a-zA-Z])_(([A-Za-z0-9]|[a-zA-Z]+)|\{([A-Za-z0-9]|[a-zA-Z]+)\})/
+
+WORD: /[a-zA-Z]+/
+
+%import common.INT -> UNSIGNED_INT
+%import common.DIGIT -> DIGIT
+%import common.SIGNED_INT -> SIGNED_INT
+%import common.SIGNED_FLOAT -> SIGNED_FLOAT
+%import common.DECIMAL -> UNSIGNED_FLOAT
+
+INT: SIGNED_INT | UNSIGNED_INT
+FLOAT: SIGNED_FLOAT | UNSIGNED_FLOAT
+
+//////////////////// grammar //////////////////////
+
+latex_string: relation | expression
+
+oneletter_symbol: SYMBOL | SUBSCRIPTED_SYMBOL
+multiletter_symbol: CMD_MATHIT L_BRACE WORD R_BRACE
+number: INT | FLOAT
+infinity: CMD_INFTY
+
+atomic_expr: oneletter_symbol
+    | multiletter_symbol
+    | number
+    | infinity
+
+atom: atomic_expr
+
+group_round_parentheses: L_PAREN expression R_PAREN
+group_square_brackets: L_BRACKET expression R_BRACKET
+group_curly_parentheses: L_BRACE expression R_BRACE
+
+relation: expression (EQUAL | NOT_EQUAL | LT | LTE | GT | GTE) expression
+
+expression_core: atom
+  | group_round_parentheses
+  | group_curly_parentheses
+
+add: expression ADD expression
+sub: expression SUB expression
+mul: expression MUL_SYMBOL expression
+    | implicit_multiplication
+div: expression DIV_SYMBOL expression
+
+implicit_multiplication: (oneletter_symbol oneletter_symbol | number oneletter_symbol)
+    | group_round_parentheses (group_round_parentheses | oneletter_symbol)
+    | function function
+
+expression: expression_core
+    | add | sub | mul | div
+    | unary_neg
+    | superscript
+    | fraction
+    | binomial
+    | function
+    | ambiguous_sub
+    | integral
+    | summation
+    | product
+//    | limit
+
+//limit: FUNC_LIM UNDERSCORE group_curly_parentheses_lim expression
+
+//group_curly_parentheses_lim: L_BRACE atomic_expr LIM_APPROACH_SYM  R_BRACE
+
+unary_neg: SUB expression
+
+integral: FUNC_INT expression "d"oneletter_symbol
+    | FUNC_INT (UNDERSCORE (group_curly_parentheses | atomic_expr) CARET (group_curly_parentheses | atomic_expr))? expression? "d"oneletter_symbol
+    | FUNC_INT (CARET (group_curly_parentheses | atomic_expr) UNDERSCORE (group_curly_parentheses | atomic_expr))? expression? "d"oneletter_symbol
+
+group_curly_parentheses_special: L_BRACE (relation | atomic_expr) R_BRACE
+
+summation: FUNC_SUM UNDERSCORE group_curly_parentheses_special CARET (group_curly_parentheses | atomic_expr) expression
+    | FUNC_SUM CARET (group_curly_parentheses | atomic_expr) UNDERSCORE group_curly_parentheses_special expression
+
+product: FUNC_PROD UNDERSCORE group_curly_parentheses_special CARET (group_curly_parentheses | atomic_expr) expression
+    | FUNC_PROD CARET (group_curly_parentheses | atomic_expr) UNDERSCORE group_curly_parentheses_special expression
+
+superscript.1: expression_core CARET expression
+
+//subscript.1: expression_core UNDERSCORE expression
+
+ambiguous_sub: oneletter_symbol number
+
+fraction: basic_fraction
+    | simple_fraction
+    | general_fraction
```

Comment:
what about just having:
```
fraction: CMD_FRAC expression_core expression_core
```
that is, `expression_core` should contain all elements that have higher priority than `\frac`.

Reply:
Setting aside the `group_round_parentheses` discussion, that would still not work. If we make this change, then in `\frac12`, the `12` would parse as `number`, when we need it to parse as `DIGIT DIGIT` in this case.

A similar thing would happen for `\frac1234` whereas we need the `1234` to parse as `DIGIT DIGIT expression` where the expression contains `34` as a `number`.